protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -1068,7 +1093,8 @@ impl ProxyExtendedChannelFactory {
         m: &SetNewPrevHashFromTp<'static>,
     ) -> Result<Option<PartialSetCustomMiningJob>, Error> {
         if let Some(job_creator) = self.job_creator.as_mut() {
-            let job_id = job_creator.on_new_prev_hash(m).unwrap_or(0);
+            // since we have staged phash 0 is no longer reserved and 1 is the default
+            let job_id = job_creator.on_new_prev_hash(m).unwrap_or(1);
```

Comment:
While we are in this file can you fix the below warnings (line 1422,1423) - i see them on so many PRs.